### ISSUE TITLE
FreeBSD Almquist Shell Security Check

### ISSUE THREAD

**Issue by probonopd** [jart/cosmopolitan#23]
```
sh  /home/user/Desktop/hello.com 

exec: /home/user/Desktop/hello.com: Exec format error
```

```
uname -a
FreeBSD FreeBSD 12.1-RELEASE-p8 FreeBSD 12.1-RELEASE-p8 GENERIC  amd64
```

**Comment by jart** [comment]
It works fine on FreeBSD for me.

![image](https://user-images.githubusercontent.com/49262/103338508-7adca480-4a33-11eb-8f24-678370d99b66.png)

Could you share the output of `readelf -Wa hello.com`? Could you try the executable at https://justine.lol/hello.com? I need to be able to reproduce this error.

**Comment by jart** [comment]
Are you using the FreeBSD fork of the Almquist Shell as your login shell? There's a known issue where @jillest added a security check that breaks compatibility with the Thompson Shell. See: https://github.com/freebsd/freebsd/commit/a81357fbe93ddb2be80ff1401450234c412c135d I've contacted jilles@freebsd.org in the past and asked him to restore the backwards compatibility, or at the very least add an exception for Actually Portable Executables which always start with the prefix "MZqFpD". Could you please email him and leave feedback? Until then, I would recommend using FreeBSD bash as your login shell, which works fine.

**Comment by probonopd** [comment]
`/bin/sh` on `FreeBSD 12.1-RELEASE-p8`. I do not know how to identify the shell further, as `sh --version` and `sh --help` do not work.

**Comment by jillest** [comment]
In https://github.com/freebsd/freebsd-src/commit/e0f5c1387df23c8c4811f5b24a7ef6ecac51a71a I adjusted the check to allow for recent POSIX changes and for Actually Portable Executables. This will become available in FreeBSD 13.0-RELEASE; I might backport it to stable/12.

Note that I would not call this a security check but a safety check. In particular, executing an ELF binary as a shell script often writes to files with strange names (because of a `>` that tends to occur).

**Comment by jart** [comment]
Thanks @jillest. That's great news! Backporting to FreeBSD 12 would be appreciated since that means Cosmopolitan will be able to fully support FreeBSD 12+ (rather than FreeBSD 13+) going forward. Helping cosmonauts (cosmopolitan users) reach the broadest audience of installs possible is part of the mission. For example, this project supports all Linux distros running Linux Kernel 2.6+ as well as Windows NT versions greater than Vista.
